src/main/webapp/src/components/youTubeRecommendations.js

```diff
@@ -81,16 +81,28 @@ class YouTubeRecommendations extends React.Component {
   componentDidUpdate(prevProps, prevState) {
     /* If state has changed, send it to songInfo component */
     if (!objectEquals(prevState, this.state)) {
-      this.props.onChangeState(this.state);
+      this.props.onChangeState(this.state, () => {});
     }
-    /* If entityState has changed, change the component state based on new entityState */
-    if (!objectEquals(prevProps.entityState, this.props.entityState)) {
-        this.getVideos(this.props.entityState);
+    /**
+     * If entityState has changed and videoIds wasn't loaded before,
+     * change the component state based on new entityState
+     */
+    if (!this.props.sentInfo.wasSent
+        && !objectEquals(prevProps.entityState, this.props.entityState)) {
+      this.getVideos(this.props.entityState);
     }
   }
 
+  /** 
+   * Get youtube video ids if they were sent before
+   * or load it instead.
+   */
   componentDidMount() {
-    this.getVideos(this.props.entityState);
+    if (this.props.sentInfo.wasSent) {
+      this.setState({ videoIds: this.props.sentInfo.info });
+    } else {
+      this.getVideos(this.props.entityState);
+    }
```

Comment:
Since `componentDidMount` is a method from the react lifeCycle, let's make it simply call another method instead of having a non-trivial logic. That might not make too much of a difference now, but it will if you want to do it expand the logic in the future. Then you can move the documentation to this method. The same was done, for instance, in `sentimentAnalysisInfo.js`